neuralprophet/forecaster.py

```diff
@@ -1711,8 +1735,412 @@ def plot_latest_forecast(
                 line_per_origin=True,
             )
 
+    def check_if_configured(self, components, error_flag=False):
```

Comment:
Why was this function moved into `forecaster.py`? To me this is all purely plotting related and would better remain in `plot_forecast.py`

Or if we want to turn this into a generic implementation for checking whether a component is configured, it would be better to allow validation of each component individually and separate the configuration check from the plotting parameter generation - guess all only relevant for another PR, I was surprised by the amount of change here.

Reply:
@noxan ok, I see. Would you recommend moving them to `plot_forecaster.py`?  I agree, with you to keep the plotting-related functions separated in order to not overload `forecaster.py`. Since the `check_if_configured()` is used for components and parameters, and the respective functions are in different .py files, is the most proper way to define the function in the `plot_forecaster.py`?

Reply:
I think it might be best to create a new file for code which is shared across our two plotting libraries. We could name it `plot_forecaster_shared.py` or something 🤔 

Reasoning: I would prefer to put it  in a file named `plot_forecaster.py`. But currently `plot_forecaster.py` is implicitly the `matplotlib` implementation, in contrast to the explicit `plotly` implementation which has the proper prefix in the filename. So this does not make much sense.
Future vision: we rename the current `plot_forecaster.py` to `plot_forecaster_matplotlib.py` and name the file for the shared code properly `plot_forecaster.py`